c10/cuda/impl/CUDAGuardImpl.h

```diff
@@ -40,6 +40,13 @@ struct CUDAGuardImpl final : public c10::impl::DeviceGuardImplInterface {
   }
   Device getDevice() const override {
     DeviceIndex device = 0;
+
+    // For supporting tracing cuda model on cuda-less environment,
+    // if device is not available, use cuda:0 by default
+    if (c10::cuda::device_count() == 0) {
+      return Device(DeviceType::CUDA, 0);
+    }
```

Comment:
Are these changes needed? If there's no cuda device wouldn't this file not be built?